### PR TITLE
Initial example app commit

### PR DESCRIPTION
**Pull request by srtena**
This PR adds:

- The initial example app code with 3 stories: **dynamic tiling**, **dynamic tiling + SQL parameters**, and **pre-generated tilesets**
- Fully functional code including a public API Access Token in the `.env`
- A `readme.md` with instructions (we need to update it when the blog comes live)

TO-DO

- [x] Initial commit
- [x] Code review
- [x] Deploy using Vercel/etc
- [ ] Update blogpost draft
- [ ] Publish blogpost
- [ ] Update readme.md

### CODE REVIEW

**Comment by srtena** [comment]
@alasarr currently HTML is managed as strings in the `main.ts` and `stories.ts` file. Should I manage it with HTML + CSS instead? Not sure why I started like this, not sure what approach you prefer 👀

File: .github/workflows/deploy.yml [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-node@v2
+        with:
+          node-version: ${{ env.NODE_VERSION }}
+
+      # Update yarn /npm commands based on your needs
+      - run: npm install
```
Comment (alasarr) [inline .github/workflows/deploy.yml:23]: All the frontend team and deck use yarm. So, my advise is to change it to yarn

File: src/map.ts [hunk 2]
```diff
@@ -0,0 +1,244 @@
...
+  render();
+}
+
+// Filter the query based on the selected year
+
+let selectedYear = 2010;
+
+export function selectYear(year: number) {
+  selectedYear = year;
+  console.log(selectedYear);
```
Comment (alasarr) [inline src/map.ts]: TO remove

File: src/map.ts [hunk 3]
```diff
@@ -0,0 +1,244 @@
...
+  render();
+}
+
+// A longer query to aggregate points (accidents) in polygons (states)
+
+const countAccidentsPerState = `
```
Comment (alasarr) [inline src/map.ts]: I would rename it to countAccidentsPerStateQuery. 

Check this, a simplified version of your query (keeping 0 values):

```sql
  SELECT
        polygons.code_hasc poligonid, any_value(polygons.geom) as geom,any_value(name) as state_name,
        count(points.year) count
      FROM `carto-demo-data.demo_tables.usa_states_boundaries` as polygons
      LEFT JOIN `carto-demo-data.demo_tables.riskanalysis_railroad_accidents` as points
        ON  ST_INTERSECTS(polygons.geom, points.geom) AND year=2010
```

NIT 1: Using the ON clause, the null values will arrive to count
NIT 2: count based on an attribute returns 0 on null values.

File: src/map.ts [hunk 4]
```diff
@@ -0,0 +1,244 @@
...
+          attr: 'group_name',
+          domain: [selectedOsmCategory],
+          colors: [[255, 255, 255]],
+          othersColor: [3, 111, 226, 0]
+        }),
+        getPointRadius: 50,
+        getLineWidth: 10,
+        pointRadiusMinPixels: 1,
+        lineWidthMinPixels: 0.3,
+        visible: selectedStory === "dynamicTiling",
```
Comment (alasarr) [inline src/map.ts]: In general you're using single quote, I'd move it  to single everything

File: src/map.ts [hunk 5]
```diff
@@ -0,0 +1,244 @@
...
+  render();
+}
+
+
+// Our render function: it sets the deck.gl layers based on our inputs
+
+export function render() {
+
+  deck.setProps({
+    layers: [
```
Comment (alasarr) [inline src/map.ts]: Just for debugging purpose I'd do:

```
const layers = [...]
deck.setProps({ layers})
```

File: src/stories.ts [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+
+    storyCard!.innerHTML = tilesetStory;
+    storyCard!.classList.add("green-bg");
+    setupCityButtonsList(document.querySelectorAll<HTMLButtonElement[]>(".city-button-group button"));
+    
+  }
+
+  selectStoryMap(value);
+
+}
```
Comment (alasarr) [inline src/stories.ts:108]: And an extra line

File: src/inputs.ts [hunk 7]
```diff
@@ -0,0 +1,22 @@
+import { selectOsmCategory, selectYear, moveToCity } from "./map";
```
Comment (alasarr) [inline src/inputs.ts]: No strong opinion, but I feel this module makes the readability of the code a little bit harder. I would move it to stories.ts. In general, you use it only once.

**Review by alasarr — CHANGES_REQUESTED** [review]
Looks great man! Just a couple of minors comments and an improvement of the SQL query

**Comment by alasarr** [comment]
>@alasarr currently HTML is managed as strings in the main.ts and stories.ts file. Should I manage it with HTML + CSS instead? Not sure why I started like this, not sure what approach you prefer 👀

I'm fine with your approach

File: src/map.ts (hunk 2 above)
Comment (srtena) [reply src/map.ts]: Done ✅

File: src/map.ts (hunk 3 above)
Comment (srtena) [reply src/map.ts]: Yeah, agree on both suggestions. I think I started using SQL from a Workflow as a basis and then never checked back to make it better. Thanks! This is the final query after grouping:

```sql

SELECT
    polygons.code_hasc as polygonid, any_value(polygons.geom) as geom,any_value(name) as state_name,
    count(points.year) count
  FROM `carto-demo-data.demo_tables.usa_states_boundaries` as polygons
  LEFT JOIN `carto-demo-data.demo_tables.riskanalysis_railroad_accidents` as points
    ON  ST_INTERSECTS(polygons.geom, points.geom) AND year=@selectedYear
  GROUP BY polygons.code_hasc

```

File: src/map.ts (hunk 4 above)
Comment (srtena) [reply src/map.ts]: ✅

File: src/map.ts (hunk 5 above)
Comment (srtena) [reply src/map.ts]: ✅ thanks!

File: src/stories.ts (hunk 6 above)
Comment (srtena) [reply src/stories.ts:108]: ✅

File: src/inputs.ts (hunk 7 above)
Comment (srtena) [reply src/inputs.ts]: Sounds fair. I did that and added a few comments to stories.ts to increase legibility

File: .github/workflows/deploy.yml (hunk 1 above)
Comment (srtena) [reply .github/workflows/deploy.yml:23]: Ready ✅  I see @raulvr2 also changed the deploy 👍 

Btw @alasarr I started from https://docs.carto.com/carto-for-developers/guides/build-a-public-application that uses NPM... Do you want to change that one too?

Comment (alasarr) [reply .github/workflows/deploy.yml:23]: No for that is right. Keep it

File: src/locations.ts [hunk 8]
```diff
@@ -0,0 +1,20 @@
...
+    longitude: 139.7203199
+  },
+  barcelona: {
+    latitude: 41.3974343,
+    longitude: 2.1610199
+  }
+}
+
+export function getLocation(city: string) { 
+  return locations[city];
```
Comment (alasarr) [inline src/locations.ts:22]: You have a type error here

File: src/map.ts [hunk 9]
```diff
@@ -0,0 +1,230 @@
...
+let deck : Deck;
+
+// Render the right layers and initial view state each time a story is selected
+
+let selectedStory = 'dynamicTiling';
+
+export function selectStoryMap(story: string) {
+  selectedStory = story;
+  deck.setProps({
+    initialViewState: INITIAL_VIEW_STATE[story]
```
Comment (alasarr) [inline src/map.ts:68]: Same here

File: index.html [hunk 10]
```diff
@@ -0,0 +1,18 @@
+<!doctype html>
+<html lang="en">
+  <head>
+    <meta charset="UTF-8" />
+    <link rel="icon" type="image/svg+xml" href="/vite.svg" />
```
Comment (alasarr) [inline index.html]: I would link to an external file and remove vit.svg file. https://github.com/CartoDB/viz-doc/blob/master/guides/public-app/index.html#L5

**Review by alasarr — DISMISSED** [review]

**Comment by alasarr** [comment]
Typescript.svg file is not required

File: index.html (hunk 10 above)
Comment (srtena) [reply index.html]: Done ✅

File: src/locations.ts (hunk 8 above)
Comment (srtena) [reply src/locations.ts:22]: Solved ✅

File: src/map.ts (hunk 9 above)
Comment (srtena) [reply src/map.ts:68]: Solved ✅

**Comment by srtena** [comment]
@alasarr I requested re-review. I learnt a bit of Typescript in the way (first time here), so let me know if you disagree with my fixes or would recommend better approaches

@raulvr2 can you help with the deploy/preview actions? Now the build is passing but Firebase can't find `index.html`. I tried with a different destination folder but I think you'll solve it quickly

**Review by alasarr — APPROVED** [review]

**Comment by srtena** [comment]
Merging 🚀

**Pull request merged**